DirectXTex/DirectXTexImage.cpp

```diff
@@ -567,6 +567,78 @@ HRESULT ScratchImage::InitializeFromImage(const Image& srcImage, bool allow1D, C
     return S_OK;
 }
 
+_Use_decl_annotations_
+HRESULT ScratchImage::InitializeSwizzleFromImage(const Image& srcImage, CP_FLAGS flags) noexcept
+{
+    HRESULT hr = Initialize2D(srcImage.format, srcImage.width, srcImage.height, 1, 1, flags);
+
+    if (FAILED(hr))
+        return hr;
+
+    uint8_t* sptr = srcImage.pixels;
+    if (!sptr)
+        return E_POINTER;
+
+    uint8_t* dptr = m_image[0].pixels;
+    if (!dptr)
+        return E_POINTER;
+
+    size_t bytesPerPixel = BitsPerPixel(srcImage.format) / 8;
```

Comment:
I've been reading up on this.  https://learn.microsoft.com/en-us/windows/win32/direct3d10/d3d10-graphics-programming-guide-resources-block-compression

Block Compression works on 4x4 blocks of pixels.  Since pixels are smushed together, we really can't swizzle them in this format.  The only solution I thought of is to Decompress them, [un]swizzle them, and re-Compress them.

However! our Compress function needs TEX_COMPRESS_FLAGS and a threshold.  Since the source image doesn't store them, I'm not sure where to get good values.  Any Ideas?

Reply:
You can just treat each block as if it were an 8byte or 16byte pixel. That's what Standard Swizzle is doing for BC.

Reply:
To clarify the Compressed case is done by treating it as a 1/4th image where each block is a "pixel".

```
size_t blockBytes = (format == DXGI_FORMAT_BC1 || format == DXGI_FORMAT_BC4) ? 8 : 16;
size_t nbw = std::max<size_t>(1, (width + 3) / 4); 
size_t nbh = std::max<size_t>(1, (height + 3) / 4);
for(size_t y = 0; y < nbh; ++y)
{
    for(size_t x = 0; x < nbw; ++x)
    {

    }
}
```

Reply:
I really need to add a helper for getting the 8 or 16 value from a format for BC. I'll fix that shortly.

Reply:
Isn't that what I was doing originally?

Reply:
If you merge the latest main, there is a new ``BytesPerBlock`` helper.